lib/nifty_services/base_service.rb

```diff
@@ -188,7 +188,9 @@ def error!(status, message_key, options = {})
     end
 
     def valid_object?(record, expected_class)
-      record.present? && record.is_a?(expected_class)
+      blank = record.respond_to?(:empty?) ? !!record.empty? : !record
```

Comment:
You're right, we can remove this! No need to check if is empty, just check if is a given class and move forward